src/tests/krate/publish.rs

```diff
@@ -202,6 +228,34 @@ fn new_krate_with_dependency() {
     assert_eq!(dependencies[0].req, "1.0.0");
 }
 
+#[test]
+fn new_krate_with_underscore_name_dependency() {
+    use super::dependencies::Deps;
+
+    let (app, anon, user, token) = TestApp::full().with_token();
+
+    app.db(|conn| {
+        CrateBuilder::new("_foo-dep", user.as_model().id).expect_build(conn);
+    });
+
+    let dependency = DependencyBuilder::new("_foo-dep").version_req("1.0.0");
+
+    let crate_to_publish = PublishBuilder::new("new_dep")
+        .version("1.0.0")
+        .dependency(dependency);
+
+    token.enqueue_publish(crate_to_publish).good();
```

Comment:
`_foo-dep` is not a legal crate name, so this call should actually fail, because in this case we were not renaming the dependency

Reply:
Oh, I see what you mean! _ can only be used in renaming.

Reply:
yeah, exactly

Reply:
I'm not 100% sure, but I think `explicit_name_in_toml: Option<EncodableCrateName>` is the issue, and maybe it should be `explicit_name_in_toml: Option<String>` instead, or at least some type that does allow underscore prefixes.

Reply:
Yes, you are right! I have added it. I'll sort out the code and submit it tomorrow!